x-pack/platform/plugins/shared/streams/server/routes/internal/streams/processing/simulation_handler.ts

```diff
@@ -815,18 +824,42 @@ const computeDetectedFields = async (
 
   const confirmedValidDetectedFields = computeMappingProperties(params.body.detected_fields ?? []);
 
-  return uniqueFields.map((name) => {
-    const existingField = streamFields[name];
-    if (existingField) {
-      return { name, ...existingField };
-    }
-
-    const existingFieldCaps = Object.keys(streamFieldCaps[name] || {});
+  return Promise.all(
+    uniqueFields.map(async (name) => {
+      const existingField = streamFields[name];
+      if (existingField) {
+        return { name, ...existingField };
+      }
 
-    const esType = existingFieldCaps.length > 0 ? existingFieldCaps[0] : undefined;
+      const existingFieldCaps = Object.keys(streamFieldCaps[name] || {});
+      const esType = existingFieldCaps.length > 0 ? existingFieldCaps[0] : undefined;
+
+      let suggestedType: string | undefined;
+      let source: string | undefined;
+      let description: string | undefined;
+
+      try {
+        const fieldMetadata = await fieldsMetadataClient.getByName(name);
```

Comment:
should we use the bulk `find()` method instead ?

Reply:
yeah, I'm gonna refactor